From dba9059db28cb8e474ae5d098a34a1c7b92d4861 Mon Sep 17 00:00:00 2001
From: Guy Nisan <gnisan@habana.ai>
Date: Tue, 10 Aug 2021 11:23:12 +0300
Subject: [PATCH] [SW-0] preboot: fix buildversion-habanalabs

When attempting to create buildversion-habanalabs we encounter an error:
In file included from ./include/preboot.h:11:0,
                 from <command-line>:32:
./arch/gaudi/arm/common.h:10:10: fatal error: address_space.h: No such
file or directory
 #include <address_space.h>
          ^~~~~~~~~~~~~~~~~
compilation terminated.

This patch update the include path which fixes this error.

Change-Id: Id10739d76aa3dce34b348c0cf65489eca93ef55a
Signed-off-by: Guy Nisan <gnisan@habana.ai>
---

diff --git a/Makefile b/Makefile
index 26bea84..531ddeb 100644
--- a/Makefile
+++ b/Makefile
@@ -110,7 +110,7 @@
 	@date +'#define PREBOOT_TIME "%T"' >> $@
 
 buildversion-habanalabs: include/preboot.h $(AUTO_GENERATED_FILE)
-	echo "VERSION_STR" | gcc -include $< -I./ -I ./lib/common/ -I ./$(HL_DEV_ARCH)/ -I$(SPECS_DIR)/$(BASE_DEV_NAME)/asic_reg -I$(SPECS_DIR)/ -I ./include/ -P -E -o - - | tail -1 | sed 's/\" \+\"//g;s/["\n]//g' | cut -d' ' -f2- | tr -d '\n' > $@
+	echo "VERSION_STR" | gcc -include $< -I./ -I ./lib/common/ -I ./$(HL_DEV_ARCH)/ -I$(SPECS_EMBEDDED_DIR)/$(SPECS_EMB_DEV) -I$(SPECS_DIR)/$(BASE_DEV_NAME)/asic_reg -I$(SPECS_DIR)/ -I ./include/ -P -E -o - - | tail -1 | sed 's/\" \+\"//g;s/["\n]//g' | cut -d' ' -f2- | tr -d '\n' > $@
 
 $(PCI_PHY_FW_TARGET): $(HL_DEV_ARCH)/pcie_phy_fw.i
 	@cut -d' ' -f2- $< | xxd -r -p | dd status=none bs=16 count=$$(($(FW_SIZE) / 16)) conv=sync,swab of=$@
